WIP Dev 34 latest update from gustav code

oleschwen opened the pull request:
Not yet ready to merge.
Will require comparison to external code.

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/data/datasets/dataset_3d_odelia.py:
```diff
@@ -0,0 +1,162 @@
+
+
+class ODELIA_Dataset3D(data.Dataset):
+    PATH_ROOT = Path('/')
```
Should this be /data (as mounted in the Docker container)?

Ultimate-Storm replied:
Yes right

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/models/__init__.py:
```diff
@@ -6,11 +6,8 @@
+from .mst import MST, MSTRegression
 
-__all__ = ['VeryBasicModel', 'BasicModel', 'BasicClassifier', 'ResNet', 'DenseNet121', 'EfficientNet', 'UNet3D', 'VisionTransformer']
+__all__ = ['VeryBasicModel', 'BasicModel', 'BasicClassifier', 'ResNet']
```
Should MST (and possibly other classes) be listed here as well?

Ultimate-Storm replied:
I have included them now

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/data/datasets/__init__.py:
```diff
@@ -2,9 +2,10 @@
-from .dataset_3d_collab import DUKE_Dataset3D_collab
-from .dataset_3d_duke import DUKE_Dataset3D
-from .dataset_3d_duke_external import DUKE_Dataset3D_external
+#from .dataset_3d import SimpleDataset3D
```
Why are most lines commented out in this file?

oleschwen replied:
OK, let's consider removing the unused classes as part of a follow-up issue.

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/env_config.py:
```diff
@@ -48,29 +41,46 @@ def prepare_dataset(task_data_name, data_dir, site_name):
     return os.path.join(scratch_dir, f"{current_time}_{task_data_name}_{model_name}_{mode}")
 
+
 def cal_weightage(train_size):
```
Method is unused and hard-codes information about Duke dataset. Remove?

Ultimate-Storm replied:
I would like to make it as an inline todo

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/env_config.py:
```diff
@@ -48,29 +41,46 @@ def prepare_dataset(task_data_name, data_dir, site_name):
+    return min(weightage, 100)
+
 
 def cal_max_epochs(preset_max_epochs, weightage):
```
Method is unused. Remove?

Ultimate-Storm replied:
Ok

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/models/base_model.py:
```diff
@@ -1,269 +1,244 @@
+        return [optimizer]
 
 
 class BasicClassifier(BasicModel):
```
Differs from source, need to take a closer look at the differences.

Ultimate-Storm commented on application/jobs/3dcnn_ptl/app/custom/data/augmentation/augmentations_3d.py:
```diff
@@ -1,244 +1,238 @@
+    }.get(model)
+
+
+class TransformerEncoder(Encoder):
```
Yes right, this was by mistake. Should be in mst.py

Ultimate-Storm commented on application/jobs/3dcnn_ptl/app/custom/env_config.py:
```diff
@@ -48,29 +41,46 @@ def prepare_dataset(task_data_name, data_dir, site_name):
+    parser.add_argument('--task', type=str, default="binary", choices=['binary', 'ordinal'])
+    parser.add_argument('--config', type=str, default="unilateral", choices=['original', 'unilateral'])
+    args = parser.parse_args()
+    binary = args.task == 'binary'
```
I changed the name here to is_binary_task

oleschwen commented:
Prior to merging, we should
- [ ] increase the version number

oleschwen commented on application/jobs/3dcnn_ptl/app/custom/threedcnn_ptl.py:
```diff
@@ -1,37 +1,29 @@
-    else:
-        MAX_EPOCHS = 5
-
+    NUM_EPOCHS_FOR_SITE = {
```
yes, see #60

oleschwen commented on README.md:
```diff
@@ -77,21 +77,64 @@ A VPN is necessary so that the swarm nodes can communicate with each other secur
+| `NUM_EPOCHS`         | `1` (test mode)| Number of training epochs (used in preflight/local training)               |
+| `TRAINING_MODE`      | derived        | Internal use. Automatically set based on flags like `--start_client`       |
+
+These are injected into the container as `--env` variables. You can modify their defaults by editing `docker.sh` or exporting before run:
```
For non-developers, these are implementation details. We should be careful not to encourage modifying the docker.sh scripts, this defeats the purpose of providing startup kits that ensure everyone runs the same code.

oleschwen commented on README.md:
````diff
@@ -75,23 +75,66 @@ A VPN is necessary so that the swarm nodes can communicate with each other secur
+     ```
+   - **Best checkpoint** usually saved as `best.ckpt` or `last.ckpt`
+   - **Prediction results**, if enabled, will appear in subfolders of the same directory
+   - **TensorBoard or WandB logs**, if activated, are stored in their respective folders inside the run directory
````
As part of #68, limit this to what is actually being logged, i.e., no WandB.

oleschwen commented:
> Not yet ready to merge. Will require comparison to external code.

Original code has been boiled down substantially, training performance looks good enough to continue.

oleschwen commented:
Summary of PR:

- integrated of Gustav’s training code boiled down to three-class classification task on ODELIA challenge data using pre-trained MST (with ResNets as an alternative) including preparation scripts (for reference)
- remove DUKE binary classification task
- refactored integration tests and included data access preflight check and simulation mode for ternary classification task (based on synthetic dataset)
- included pre-trained weights of dinov2 in docker image
- updates of apt package versions for Docker image
- updated README and startup kits accordingly
- utility scripts for developers
